pom.xml

```diff
@@ -5,27 +5,42 @@
   <parent>
     <groupId>com.hubspot</groupId>
     <artifactId>basepom</artifactId>
-    <version>18.3</version>
+    <version>25.4</version>
   </parent>
 
   <groupId>com.hubspot.slack</groupId>
   <artifactId>slack-client</artifactId>
   <version>1.2-SNAPSHOT</version>
   <packaging>pom</packaging>
 
-  <properties>
-    <project.build.targetJdk>1.8</project.build.targetJdk>
-    <dep.protobuf-java.version>3.5.1</dep.protobuf-java.version>
-  </properties>
-
   <modules>
     <module>slack-base</module>
     <module>slack-java-client</module>
     <module>slack-java-client-examples</module>
   </modules>
 
+  <properties>
+    <dep.immutables.version>2.2.10</dep.immutables.version>
```

Comment:
Need to downgrade Immutables because newer versions don't like `SlackEvent#toDetailedEvent`:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/jhaber/src/slack-client/slack-base/src/main/java/com/hubspot/slack/client/models/events/SlackEvent.java:[11,36] Method 'toDetailedEvent' cannot have own generic type parameters. Attribute accessors can only use enclosing type's type variables
```

I've deprecated this method and we should remove it in the future (I don't see any annotation that would cause immutables to ignore it)